scripts/build-upload-docker-images.sh

```diff
@@ -0,0 +1,87 @@
+#!/bin/bash
+
+set -euxf -o pipefail
+
+build_upload_multiarch_images(){
+	for component in agent collector query ingester
+	do
+		docker buildx build --output "${PUSHTAG}" \
+			--progress=plain --target release \
+			--build-arg base_image="localhost:5000/baseimg:1.0.0-alpine-3.12" \
+			--build-arg debug_image="golang:1.15-alpine" \
+			--platform=${PLATFORMS} \
+			--file cmd/${component}/Dockerfile \
+			$(echo ${IMAGE_TAGS} | sed "s/JAEGERCOMP/${component}/g") \
+			cmd/${component}
+		echo "Finished building multiarch jager-${component} =============="
+	done
+
+	for component in es-index-cleaner es-rollover tracegen anonymizer
+	do
+		if [[ "${component}" == "es-rollover" ]]; then
+			docker_file_arg="--file plugin/storage/es/Dockerfile.rollover"
+		else
+			docker_file_arg=""
+		fi
+		if [[ "${component}" =~ ^es-.* ]]; then
```

Comment:
Won't this regex match for es-rollover too ? Is that intended ?

Reply:
This multiple if-else condition is making things confusing. Can we make it cleaner with either switch-case or some other way ?

Reply:
a simpler approach is to define a function with args like docker_file_arg and then call the function for each component with necessary args

Reply:
@Ashmita152  Yes, this regex match for both `es-index-cleaner` and `es-rollover` , since the value of their `dir_arg` is the same (`plugin/storage/es`).

Reply:
I've separated out a function docker_buildx_build() in this script and call it for each component with necessary args. Thanks.